diff --git a/tinydm-run-session.sh b/tinydm-run-session.sh
index 5f4eb69..cadf9fb 100644
--- a/tinydm-run-session.sh
+++ b/tinydm-run-session.sh
@@ -1,6 +1,12 @@
 #!/bin/sh
 # Copyright 2020 Oliver Smith
 # SPDX-License-Identifier: GPL-3.0-or-later
+#
+# Patched version of tinyDM to allow path configuration - all of the following env vars must be set:
+# TINYDM_X11_PROFILE_PATH     - path of profiles to be sourced on startup
+# TINYDM_WAYLAND_PROFILE_PATH - path of profiles to be sourced on startup
+#echo "TINYDM_X11_PROFILE_PATH is ${TINYDM_X11_PROFILE_PATH:?EMPTY}"
+#echo "TINYDM_WAYLAND_PROFILE_PATH is ${TINYDM_WAYLAND_PROFILE_PATH:?EMPTY}"
 
 setup_log() {
 	logfile=${XDG_STATE_HOME:-~/.local/state}/tinydm.log
@@ -30,7 +36,12 @@ source_session_profiles() {
 		exit 1
 	fi
 
-	for file in "/etc/tinydm.d/env-${session_type}.d/"*; do
+  profile_path=$TINYDM_X11_PROFILE_PATH
+  if [ "$session_type" == "wayland"]; then
+    profile_path=$TINYDM_WAYLAND_PROFILE_PATH
+  fi
+
+	for file in "${profile_path}"*; do
 		if ! [ -e "$file" ]; then
 			continue
 		fi
@@ -98,22 +109,33 @@ run_session() {
 	echo "Exec:    $cmd"
 	echo "---"
 
-	case "$resolved" in
-		/usr/share/wayland-sessions*)
-			source_session_profiles wayland
-			# shellcheck disable=SC2086
-			run_session_wayland $cmd
-			;;
-		/usr/share/xsessions*)
-			source_session_profiles x11
-			# shellcheck disable=SC2086
-			run_session_x $cmd
-			;;
-		*)
-			echo "ERROR: could not detect session type!"
-			exit 1
-			;;
-	esac
+  IFS=$'\n: '
+  for dir in $XDG_DATA_DIRS
+  do
+    xsessions=$(realpath ${dir}/xsessions/* 2>/dev/null)
+    wsessions=$(realpath ${dir}/wayland-sessions/* 2>/dev/null)
+    for session in ${wsessions}
+    do
+    if [ "$resolved" = "$session" ]; then
+      echo "tinydm: running wayland session ${session} (${cmd})"
+      source_session_profiles wayland
+      # shellcheck disable=SC2086
+      run_session_wayland $cmd
+      break
+    fi
+    done
+
+    for session in ${xsessions}
+    do
+      if [ "$resolved" = "$session" ]; then
+        echo "tinydm: running xorg session ${session} (${cmd})"
+        source_session_profiles x11
+        # shellcheck disable=SC2086
+        run_session_x $cmd
+        break
+      fi
+    done
+  done
 }
 
 setup_log
