diff --git a/Makefile b/Makefile
index 3c47028..e7fc460 100644
--- a/Makefile
+++ b/Makefile
@@ -81,7 +81,7 @@ install-scripts: $(PROGRAMS)
 
 	# Appscripts
 	mkdir -p "$(DESTDIR)$(PREFIX)/share/sxmo/appscripts"
-	cd scripts/appscripts && find . -name 'sxmo_*.sh' | xargs -I{} ln -fs "$(PREFIX)/bin/{}" "$(DESTDIR)$(PREFIX)/share/sxmo/appscripts/{}" && cd ../..
+	cd scripts/appscripts && find . -name 'sxmo_*.sh' | xargs -I{} ln -fs "$(DESTDIR)$(PREFIX)/bin/{}" "$(DESTDIR)$(PREFIX)/share/sxmo/appscripts/{}" && cd ../..
 
 
 	mkdir -p "$(DESTDIR)$(PREFIX)/share/superd/services"
