diff --git a/configs/default_hooks/sxmo_hook_lisgdstart.sh b/configs/default_hooks/sxmo_hook_lisgdstart.sh
index d7ba530..728fa5e 100644
--- a/configs/default_hooks/sxmo_hook_lisgdstart.sh
+++ b/configs/default_hooks/sxmo_hook_lisgdstart.sh
@@ -11,7 +11,7 @@ LISGD_THRESHOLD_PRESSED="${SXMO_LISGD_THRESHOLD_PRESSED:-60}"
 LISGD_INPUT_DEVICE="${SXMO_LISGD_INPUT_DEVICE:-"/dev/input/touchscreen"}"
 
 if [ dwm = "$SXMO_WM" ]; then
-	case "$(xrandr | grep DSI-1 | cut -d' ' -f 5)" in
+  case "$(xrandr | grep primary | cut -d' ' -f 5)" in
 		right) orientation=1;;
 		left) orientation=3;;
 		inverted) orientation=2;;
diff --git a/configs/default_hooks/sxmo_hook_start.sh b/configs/default_hooks/sxmo_hook_start.sh
index 277646a..2004d78 100644
--- a/configs/default_hooks/sxmo_hook_start.sh
+++ b/configs/default_hooks/sxmo_hook_start.sh
@@ -43,6 +43,7 @@ case "$SXMO_WM" in
 		superctl start autocutsel
 		superctl start autocutsel-primary
 		superctl start sxmo-x11-status
+    [ -n "$SXMO_MONITOR" ] && xrandr --output "$SXMO_MONITOR" --primary
 		;;
 esac
 
diff --git a/scripts/deviceprofiles/sxmo_deviceprofile_pine64,pinephone-1.2.sh b/scripts/deviceprofiles/sxmo_deviceprofile_pine64,pinephone-1.2.sh
index 2a46a79..624a262 100644
--- a/scripts/deviceprofiles/sxmo_deviceprofile_pine64,pinephone-1.2.sh
+++ b/scripts/deviceprofiles/sxmo_deviceprofile_pine64,pinephone-1.2.sh
@@ -6,3 +6,4 @@ export SXMO_WIFI_MODULE=8723cs
 export SXMO_SYS_FILES="/sys/module/$SXMO_WIFI_MODULE/parameters/rtw_scan_interval_thr /sys/power/state /sys/devices/platform/soc/1f00000.rtc/power/wakeup /sys/power/mem_sleep /sys/bus/usb/drivers/usb/unbind /sys/bus/usb/drivers/usb/bind /dev/rtc0 /sys/devices/platform/soc/1f03400.rsb/sunxi-rsb-3a3/axp221-pek/power/wakeup"
 export SXMO_TOUCHSCREEN_ID=8
 export SXMO_ALSA_CONTROL_NAME=PinePhone
+export SXMO_MONITOR="DSI-1"
diff --git a/scripts/deviceprofiles/sxmo_deviceprofile_pine64,pinephone-pro.sh b/scripts/deviceprofiles/sxmo_deviceprofile_pine64,pinephone-pro.sh
index 1d5325a..f515a6f 100644
--- a/scripts/deviceprofiles/sxmo_deviceprofile_pine64,pinephone-pro.sh
+++ b/scripts/deviceprofiles/sxmo_deviceprofile_pine64,pinephone-pro.sh
@@ -4,6 +4,7 @@
 
 export SXMO_SYS_FILES="/sys/power/state /sys/devices/platform/soc/1f00000.rtc/power/wakeup /sys/power/mem_sleep /sys/bus/usb/drivers/usb/unbind /sys/bus/usb/drivers/usb/bind /dev/rtc0 /sys/devices/platform/soc/1f03400.rsb/sunxi-rsb-3a3/axp221-pek/power/wakeup"
 export SXMO_TOUCHSCREEN_ID=8
+export SXMO_MONITOR="DSI-1"
 export SXMO_POWER_BUTTON="1:1:gpio-key-power"
 export SXMO_VOLUME_BUTTON="1:1:adc-keys"
 export SXMO_MODEMRTC=12
