diff --git a/scripts/core/sxmo_uniq_exec.sh b/scripts/core/sxmo_uniq_exec.sh
index 1618a1c..435dee0 100644
--- a/scripts/core/sxmo_uniq_exec.sh
+++ b/scripts/core/sxmo_uniq_exec.sh
@@ -2,6 +2,9 @@
 # SPDX-License-Identifier: AGPL-3.0-only
 # Copyright 2022 Sxmo Contributors
 
+# shellcheck source=scripts/core/sxmo_common.sh
+. sxmo_common.sh
+
 mkdir -p "$XDG_RUNTIME_DIR/sxmo.flock"
 LOCKER="$XDG_RUNTIME_DIR/sxmo.flock/$(realpath "$1" | sed 's|/|-|g').lock"
 
