diff --git a/scripts/modem/sxmo_modem.sh b/scripts/modem/sxmo_modem.sh
index 4c6171b..a05295f 100644
--- a/scripts/modem/sxmo_modem.sh
+++ b/scripts/modem/sxmo_modem.sh
@@ -44,7 +44,7 @@ checkforfinishedcalls() {
 		CONTACT="$(sxmo_contacts.sh --name "$FINISHEDNUMBER")"
 		[ "$CONTACT" = "???" ] && CONTACT="$FINISHEDNUMBER"
 
-		TIME="$(date +%FT%H:%M:%S%z)"
+		TIME="$(coreutils --coreutils-prog=date +%FT%H:%M:%S%z)"
 		mkdir -p "$SXMO_LOGDIR"
 		if [ -f "$XDG_RUNTIME_DIR/${FINISHEDCALLID}.discardedcall" ]; then
 			#this call was discarded
@@ -124,7 +124,7 @@ checkforincomingcalls() {
 	INCOMINGNUMBER="$(cleanupnumber "$INCOMINGNUMBER")"
 	CONTACTNAME=$(sxmo_contacts.sh --name "$INCOMINGNUMBER")
 
-	TIME="$(date +%FT%H:%M:%S%z)"
+	TIME="$(coreutils --coreutils-prog=date +%FT%H:%M:%S%z)"
 	if cut -f1 "$SXMO_BLOCKFILE" 2>/dev/null | grep -q "^$INCOMINGNUMBER$"; then
 		stderr "BLOCKED call from number: $VOICECALLID"
 		sxmo_modemcall.sh mute "$VOICECALLID"
@@ -171,7 +171,7 @@ checkfornewtexts() {
 		# SMS with no TEXTID is an SMS WAP (I think). So skip.
 		if [ -z "$TEXTDATA" ]; then
 			stderr "Received an empty SMS (TEXTID: $TEXTID).  I will assume this is an MMS."
-			printf %b "$(date +%FT%H:%M:%S%z)\tdebug_mms\tNULL\tEMPTY (TEXTID: $TEXTID)\n" >> "$SXMO_LOGDIR/modemlog.tsv"
+			printf %b "$(coreutils --coreutils-prog=date +%FT%H:%M:%S%z)\tdebug_mms\tNULL\tEMPTY (TEXTID: $TEXTID)\n" >> "$SXMO_LOGDIR/modemlog.tsv"
 			if [ -f "${SXMO_MMS_BASE_DIR:-"$HOME"/.mms/modemmanager}/mms" ]; then
 				continue
 			else
@@ -187,7 +187,7 @@ checkfornewtexts() {
 		NUM="$(cleanupnumber "$NUM")"
 
 		TIME="$(echo "$TEXTDATA" | grep sms.properties.timestamp | sed -E 's/^sms\.properties\.timestamp\s+:\s+//')"
-		TIME="$(date +%FT%H:%M:%S%z -d "$TIME")"
+		TIME="$(coreutils --coreutils-prog=date +%FT%H:%M:%S%z -d "$TIME")"
 
 		# Note: this will *not* block MMS, since we have to unpack the phone numbers for an MMS
 		# later.
