diff --git a/scripts/core/sxmo_migrate.sh b/scripts/core/sxmo_migrate.sh
index db64b9d..a87eabb 100644
--- a/scripts/core/sxmo_migrate.sh
+++ b/scripts/core/sxmo_migrate.sh
@@ -6,7 +6,7 @@
 # shellcheck source=scripts/core/sxmo_common.sh
 . sxmo_common.sh
 
-. /etc/profile.d/sxmo_init.sh
+. sxmo_init.sh
 _sxmo_load_environments
 
 smartdiff() {
diff --git a/scripts/core/sxmo_rtcwake.sh b/scripts/core/sxmo_rtcwake.sh
index 4e7dd65..7f0e71b 100644
--- a/scripts/core/sxmo_rtcwake.sh
+++ b/scripts/core/sxmo_rtcwake.sh
@@ -3,7 +3,7 @@
 # Copyright 2022 Sxmo Contributors
 
 # shellcheck source=configs/profile.d/sxmo_init.sh
-. /etc/profile.d/sxmo_init.sh
+. sxmo_init.sh
 
 . sxmo_common.sh
 
diff --git a/scripts/core/sxmo_winit.sh b/scripts/core/sxmo_winit.sh
index 74ecaa3..bf30792 100644
--- a/scripts/core/sxmo_winit.sh
+++ b/scripts/core/sxmo_winit.sh
@@ -40,7 +40,7 @@ cleanup() {
 
 init() {
 	# shellcheck source=/dev/null
-	. /etc/profile.d/sxmo_init.sh
+	. sxmo_init.sh
 
 	_sxmo_load_environments
 	_sxmo_prepare_dirs
diff --git a/scripts/core/sxmo_xinit.sh b/scripts/core/sxmo_xinit.sh
index 6f95f5b..a9d9181 100755
--- a/scripts/core/sxmo_xinit.sh
+++ b/scripts/core/sxmo_xinit.sh
@@ -65,7 +65,7 @@ cleanup() {
 
 init() {
 	# shellcheck source=/dev/null
-	. /etc/profile.d/sxmo_init.sh
+	. sxmo_init.sh
 
 	_sxmo_load_environments
 	_sxmo_prepare_dirs
