diff --git a/configs/default_hooks/sxmo_hook_start.sh b/configs/default_hooks/sxmo_hook_start.sh
index 94255f9..d23ca6f 100755
--- a/configs/default_hooks/sxmo_hook_start.sh
+++ b/configs/default_hooks/sxmo_hook_start.sh
@@ -20,11 +20,6 @@ while ! superctl status > /dev/null 2>&1; do
 	sleep 0.1
 done
 
-# Load our sound daemons
-# pipewire-pulse will start pipewire
-superctl start pipewire-pulse
-superctl start wireplumber
-
 # mako/dunst are required for warnings.
 # load some other little things here too.
 case "$SXMO_WM" in
