diff --git a/scripts/core/sxmo_bluetoothmenu.sh b/scripts/core/sxmo_bluetoothmenu.sh
index 238a055..85f3933 100755
--- a/scripts/core/sxmo_bluetoothmenu.sh
+++ b/scripts/core/sxmo_bluetoothmenu.sh
@@ -37,7 +37,7 @@ _restart_bluetooth() {
 		alpine|postmarketos)
 			doas rc-service bluetooth restart
 			;;
-		arch|archarm)
+		arch|archarm|nixos)
 			doas systemctl restart bluetooth
 			;;
 	esac
diff --git a/scripts/core/sxmo_bluetoothtoggle.sh b/scripts/core/sxmo_bluetoothtoggle.sh
index ff310af..c165b98 100755
--- a/scripts/core/sxmo_bluetoothtoggle.sh
+++ b/scripts/core/sxmo_bluetoothtoggle.sh
@@ -24,7 +24,7 @@ on() {
 			rc-service bluetooth start
 			rc-update add bluetooth
 			;;
-		arch|archarm)
+		arch|archarm|nixos)
 			systemctl start bluetooth
 			systemctl enable bluetooth
 			;;
@@ -37,7 +37,7 @@ off() {
 			rc-service bluetooth stop
 			rc-update del bluetooth
 			;;
-		arch|archarm)
+		arch|archarm|nixos)
 			systemctl stop bluetooth
 			systemctl disable bluetooth
 			;;
diff --git a/scripts/core/sxmo_timezonechange.sh b/scripts/core/sxmo_timezonechange.sh
index 659889d..efc9e6e 100755
--- a/scripts/core/sxmo_timezonechange.sh
+++ b/scripts/core/sxmo_timezonechange.sh
@@ -42,6 +42,7 @@ if [ $# -gt 0 ]; then
 	case "$OS" in
 		alpine|postmarketos) change_alpine "$@";;
 		arch|archarm) change_arch "$@";;
+		nixos) echo "Change your timezone in configuration.nix with time.timeZone = \"[timezone]\""
 		*) echo "Changing the timezone isn't implemented on your distro yet";;
 	esac
 else
diff --git a/scripts/core/sxmo_upgrade.sh b/scripts/core/sxmo_upgrade.sh
index 3824a51..4ef20db 100755
--- a/scripts/core/sxmo_upgrade.sh
+++ b/scripts/core/sxmo_upgrade.sh
@@ -24,7 +24,15 @@ update_pacman() {
 	echo "Upgrade complete - reboot for all changes to take effect"
 }
 
+update_nixos() {
+	echo "Upgrading all packages"
+	# nohup needed because nixos-rebuild might restart the display manager
+	# (and thus the terminal we're running in) before the update is complete
+	doas nohup nixos-rebuild switch --upgrade > /tmp/sxmo-last-upgrade.log & tail -f /tmp/sxmo-last-upgrade.log
+}
+
 case "$OS" in
 	alpine|postmarketos) update_apk;;
 	arch|archarm) update_pacman;;
+	nixos) update_nixos;;
 esac
diff --git a/configs/default_hooks/sxmo_hook_restart_modem_daemons.sh b/configs/default_hooks/sxmo_hook_restart_modem_daemons.sh
index 7e567f3..58ffce5 100755
--- a/configs/default_hooks/sxmo_hook_restart_modem_daemons.sh
+++ b/configs/default_hooks/sxmo_hook_restart_modem_daemons.sh
@@ -21,7 +21,7 @@ daemon_start() {
 		alpine|postmarketos)
 			doas rc-service "$1" start
 			;;
-		arch|archarm|debian)
+		arch|archarm|debian|nixos)
 			[ "$1" = "modemmanager" ] && set -- ModemManager
 			doas systemctl start "$1"
 			;;
@@ -41,7 +41,7 @@ daemon_stop() {
 		alpine|postmarketos)
 			doas rc-service "$1" stop
 			;;
-		arch|archarm)
+		arch|archarm|nixos)
 			[ "$1" = "modemmanager" ] && set -- ModemManager
 			doas systemctl stop "$1"
 			;;
@@ -57,7 +57,7 @@ daemon_isrunning() {
 		alpine|postmarketos)
 			rc-service "$1" status | grep -q started
 			;;
-		arch|archarm|debian)
+		arch|archarm|debian|nixos)
 			[ "$1" = "modemmanager" ] && set -- ModemManager
 			systemctl is-active --quiet "$1"
 			;;
@@ -69,7 +69,7 @@ daemon_exists() {
 		alpine|postmarketos)
 			[ -f /etc/init.d/"$1" ]
 			;;
-		arch|archarm|debian)
+		arch|archarm|debian|nixos)
 			[ "$1" = "modemmanager" ] && set -- ModemManager
 			systemctl status "$1" > /dev/null 2>&1
 			[ $? -ne 4 ]
