diff --git a/configs/default_hooks/sxmo_hook_start.sh b/configs/default_hooks/sxmo_hook_start.sh
index ff4bcd2..6c0fac3 100644
--- a/configs/default_hooks/sxmo_hook_start.sh
+++ b/configs/default_hooks/sxmo_hook_start.sh
@@ -75,10 +75,6 @@ superctl start sxmo_notificationmonitor
 # To setup initial lock state
 sxmo_hook_unlock.sh
 
-superctl start pipewire
-superctl start pipewire-pulse
-superctl start wireplumber
-
 (
 	sleep 5 # let some time to pipewire
 	superctl start callaudiod
