diff --git a/scripts/core/sxmo_wmtoggle.sh b/scripts/core/sxmo_wmtoggle.sh
index 9af41f0..ab29970 100644
--- a/scripts/core/sxmo_wmtoggle.sh
+++ b/scripts/core/sxmo_wmtoggle.sh
@@ -10,12 +10,12 @@ not_ready_yet() {
 case "$(realpath /var/lib/tinydm/default-session.desktop)" in
 	/usr/share/wayland-sessions/swmo.desktop)
 		command -v dwm >/dev/null || not_ready_yet
-		doas tinydm-set-session -f -s /usr/share/xsessions/sxmo.desktop
+		tinydm-set-session -f -s /usr/share/xsessions/sxmo.desktop
 		pkill sway
 		;;
 	/usr/share/xsessions/sxmo.desktop)
 		command -v sway >/dev/null || not_ready_yet
-		doas tinydm-set-session -f -s /usr/share/wayland-sessions/swmo.desktop
+		tinydm-set-session -f -s /usr/share/wayland-sessions/swmo.desktop
 		pkill dwm
 		;;
 esac
